crates/red_knot/src/salsa_db/source.rs

```diff
@@ -0,0 +1,190 @@
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use countme::Count;
+use dashmap::mapref::entry::Entry;
+use filetime::FileTime;
+
+use ruff_python_ast::{ModModule, PySourceType};
+use ruff_python_parser::Parsed;
+
+use crate::FxDashMap;
+
+/// A file can be considered unchanged for as long as it has the same revision.
+///
+/// The value of the revision itself has no meaning other than to encode the version of the file.
+/// Therefore, it's not possible to identify which revision is newer by comparing the values.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum FileRevision {
+    LastModified(FileTime),
+    #[allow(unused)]
+    ContentHash(u128),
+}
```

Comment:
No, this was mainly to explore how and if we could support file revisions e.g. based on a file's hash rather than the last modified timestamp. But this isn't used right now.